both/access_control.js

```diff
@@ -231,101 +229,75 @@ if(Meteor.isServer) {
         'createdAt',
       ];
 
-      if(_.difference(fields, allowedFields).length > 0) {
+      return _onlyAllowedFields(fields, allowedFields);
+    },
+    fetch: [ 'competitionId' ],
+  });
+
+  RoundProgresses.allow({
+    update: function(userId, progress, fields, modifier) {
+      if(getCannotManageCompetitionReason(userId, progress.competitionId)) {
         return false;
       }
-      return true;
+      var allowedFields = [
+        'done',
+        'total',
+      ];
+      return _onlyAllowedFields(fields, allowedFields);
     },
-    fetch: [ 'competitionId' ],
   });
 
   Results.allow({
     update: function(userId, result, fields, modifier) {
-      var competition = Competitions.findOne({ _id: result.competitionId }, { fields: { _id: 1 } });
-      if(getCannotManageCompetitionReason(userId, competition._id)) {
+      if(getCannotManageCompetitionReason(userId, result.competitionId)) {
         return false;
       }
-
       var allowedFields = [
         'solves',
 
         'updatedAt',
         'createdAt',
       ];
-
-      if(_.difference(fields, allowedFields).length > 0) {
-        return false;
-      }
-      return true;
+      return _onlyAllowedFields(fields, allowedFields);
     },
     fetch: [ 'competitionId' ],
   });
 
-  var allowedOwnRegistrationFields = [
-    'userId',
-    'competitionId',
-    'uniqueName',
-    'registeredEvents',
-    'guestCount',
-    'comments',
-
-    'updatedAt',
-    'createdAt',
-  ];
   Registrations.allow({
     insert: function(userId, registration) {
-      if(!getCannotManageCompetitionReason(userId, registration.competitionId)) {
-        // If you're allowed to manage the competition, then you can
-        // add any registration.
-        return true;
-      }
-      if(getCannotRegisterReasons(registration.competitionId)) {
-        return false;
-      }
-      // can only edit entries with their user id
-      if(registration.userId != userId) {
-        return false;
-      }
-
-      if(_.difference(fields, allowedOwnRegistrationFields).length > 0) {
-        return false;
-      }
-      return true;
+      return Registrations.userIsAllowed(userId, registration);
```

Comment:
It looks like this new code is not enforcing the fields in the registration doc. It also looks like the old code suffered from a bug similar to #209, lol. At some point, we're going to put payment information in the Registration collection, so we don't want to let clients populate the payment field when they do an initial insert. Again, I find it hard to reason about access control rules, and this would all might be much better if we used methods.